src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate){
+        this.isPrivate = isPrivate;
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Mentioned in class that a Contact object can be "equal" even though they are of different child classes (email and address have the same value for example). You should do the comparison checks in the child classes.